### PR TITLE
Update and remove many with action call

### PR DESCRIPTION
**Pull request by mariusbackes**
Update: Added the parameter **where** and **update**. If the are present and **id** is not present, the update many function will be called on the adapter.

Remove: Added the parameter **where**. If this parameter is present and **id** is not present, the remove many function will be called on the adapter.

### CODE REVIEW

File: packages/moleculer-db/src/index.js [hunk 1]
```diff
@@ -886,14 +887,22 @@ module.exports = {
...
+				return this.adapter.updateById(id, { "$set": sets })
+					.then(doc => {
+						if (!doc)
+							return Promise.reject(new EntityNotFoundError(id));
+						return this.transformDocuments(ctx, {}, doc)
+							.then(json => this.entityChanged("updated", json, ctx).then(() => json));
+					});
+			}
+			
+			return Promise.reject(new ValidationError("invalid_params"));
```
Comment (icebob) [inline packages/moleculer-db/src/index.js:905]: Please use more accurate errors, Correct message text, type, code...etc.

**Review by icebob — CHANGES_REQUESTED** [review]

File: packages/moleculer-db/src/index.js (hunk 1 above)
Comment (mariusbackes) [reply packages/moleculer-db/src/index.js]: I have added a more accurate error message and params as data. I have also added the missing test for failing the _update and _remove method

**Comment by icebob** [comment]
I discussed your PR on Discord chat and I have a problem with your PR. To enable clients to make removeMany command with filters via REST API is a little bit dangerous. I mean if somebody upgrades the moleculer-db, and after that, any user can execute a `DELETE /api/users/0?where={}` command, which removes all record from the users table.
On the other hand, if somebody needs this feature they can implement in his/her services file with a few lines of code which calls the `this.adapter.removeMany`. But before executing they can validate the filtering.

Personally, I prefer using more specified actions for removing, e.g. `removeInactiveUsers`, `removeUnfulfilledOrders`...etc instead of a universal where anybody can remove all rows from a Postman client.
@AndreMaz WDYT?

**Comment by AndreMaz** [comment]
@icebob I'm not a huge fan of this PR. Creating a possibility of potentially deleting every record is not a good idea. People tend to forget things meaning that someone somewhere would do this and it would cause a lot of problems.

**Comment by bzd2000** [comment]
But maybe an internal method like _updateMany or _removeMany could be interesting, because it would also manage the cache.
This can then be used in a custom action like "removeInactiveUsers"

**Comment by mariusbackes** [comment]
I have updated the remove method. It is not longer allowed to pass an empty object as filter.

**Comment by bzd2000** [comment]
But still you could pass something that is always true as a query.  I think is just too dangerous to have this kind of methods public available.

**Comment by icebob** [comment]
Yeah @bzd2000 is right. You can make, e.g. `{ status: 1 }` which will delete all active records. 
@mariusbackes Thank you for your work, but I don't like to merge it, sorry.

**Comment by mariusbackes** [comment]
What if I change the code, and remove the option from the remove, would it be possible to use it for the update function?

**Comment by 0x0a0d** [comment]
why not spit to `removeMany` action without set default `rest`?
I think this could be
```js
{
  actions: {
    removeMany: {
      params: {
        id: 'array'
      },
      handler() {
        //
      }
    }
  }
}
```

**Pull request closed without merging**